File: decompiled/sources/com/hjq/base/BaseAdapter.java
package com.hjq.base;

import android.content.Context;
import android.content.res.Resources;
import android.graphics.drawable.Drawable;
import android.util.SparseArray;
import android.view.LayoutInflater;
import android.view.View;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import com.hjq.base.BaseAdapter.ViewHolder;
import com.hjq.base.action.ResourcesAction;

public abstract class BaseAdapter<VH extends BaseAdapter<?>.ViewHolder> extends RecyclerView.Adapter<VH> implements ResourcesAction {
    /* access modifiers changed from: private */
    public SparseArray<OnChildClickListener> mChildClickListeners;
    /* access modifiers changed from: private */
    public SparseArray<OnChildLongClickListener> mChildLongClickListeners;
    private final Context mContext;
    /* access modifiers changed from: private */
    public OnItemClickListener mItemClickListener;
    /* access modifiers changed from: private */
    public OnItemLongClickListener mItemLongClickListener;
    /* access modifiers changed from: private */
    public int mPositionOffset = 0;
    /* access modifiers changed from: private */
    public RecyclerView mRecyclerView;

    public interface OnChildClickListener {
        void onChildClick(RecyclerView recyclerView, View view, int i);
    }

    public interface OnChildLongClickListener {
        boolean onChildLongClick(RecyclerView recyclerView, View view, int i);
    }

    public interface OnItemClickListener {
        void onItemClick(RecyclerView recyclerView, View view, int i);
    }

    public interface OnItemLongClickListener {
        boolean onItemLongClick(RecyclerView recyclerView, View view, int i);
    }

    public /* synthetic */ int getColor(int i) {
        return ResourcesAction.CC.$default$getColor(this, i);
    }

    public /* synthetic */ Drawable getDrawable(int i) {
        return ResourcesAction.CC.$default$getDrawable(this, i);
    }

    public /* synthetic */ Resources getResources() {
        return ResourcesAction.CC.$default$getResources(this);
    }

    public /* synthetic */ String getString(int i) {
        return ResourcesAction.CC.$default$getString(this, i);
    }

    public /* synthetic */ String getString(int i, Object... objArr) {
        return ResourcesAction.CC.$default$getString(this, i, objArr);
    }

    public /* synthetic */ Object getSystemService(Class cls) {
        return ResourcesAction.CC.$default$getSystemService(this, cls);
    }

    public BaseAdapter(Context context) {
        this.mContext = context;
        if (context == null) {
            throw new IllegalArgumentException("are you ok?");
        }
    }

    public final void onBindViewHolder(VH vh, int i) {
        this.mPositionOffset = i - vh.getAdapterPosition();
        vh.onBindView(i);
    }

    public RecyclerView getRecyclerView() {
        return this.mRecyclerView;
    }

    public Context getContext() {
        return this.mContext;
    }

    public abstract class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener, View.OnLongClickListener {
        public abstract void onBindView(int i);

        public ViewHolder(BaseAdapter baseAdapter, int i) {
            this(LayoutInflater.from(baseAdapter.getContext()).inflate(i, baseAdapter.mRecyclerView, false));
        }

        public ViewHolder(View view) {
            super(view);
            if (BaseAdapter.this.mItemClickListener != null) {
                view.setOnClickListener(this);
            }
            if (BaseAdapter.this.mItemLongClickListener != null) {
                view.setOnLongClickListener(this);
            }
            if (BaseAdapter.this.mChildClickListeners != null) {
                for (int i = 0; i < BaseAdapter.this.mChildClickListeners.size(); i++) {
                    View findViewById = findViewById(BaseAdapter.this.mChildClickListeners.keyAt(i));
                    if (findViewById != null) {
                        findViewById.setOnClickListener(this);
                    }
                }
            }
            if (BaseAdapter.this.mChildLongClickListeners != null) {
                for (int i2 = 0; i2 < BaseAdapter.this.mChildLongClickListeners.size(); i2++) {
                    View findViewById2 = findViewById(BaseAdapter.this.mChildLongClickListeners.keyAt(i2));
                    if (findViewById2 != null) {
                        findViewById2.setOnLongClickListener(this);
                    }
                }
            }
        }

        /* access modifiers changed from: protected */
        public final int getViewHolderPosition() {
            return getLayoutPosition() + BaseAdapter.this.mPositionOffset;
        }

        public void onClick(View view) {
            OnChildClickListener onChildClickListener;
            int viewHolderPosition = getViewHolderPosition();
            if (viewHolderPosition >= 0 && viewHolderPosition < BaseAdapter.this.getItemCount()) {
                if (view == getItemView()) {
                    if (BaseAdapter.this.mItemClickListener != null) {
                        BaseAdapter.this.mItemClickListener.onItemClick(BaseAdapter.this.mRecyclerView, view, viewHolderPosition);
                    }
                } else if (BaseAdapter.this.mChildClickListeners != null && (onChildClickListener = (OnChildClickListener) BaseAdapter.this.mChildClickListeners.get(view.getId())) != null) {
                    onChildClickListener.onChildClick(BaseAdapter.this.mRecyclerView, view, viewHolderPosition);
                }
            }
        }

        public boolean onLongClick(View view) {
            OnChildLongClickListener onChildLongClickListener;
            int viewHolderPosition = getViewHolderPosition();
            if (viewHolderPosition >= 0 && viewHolderPosition < BaseAdapter.this.getItemCount()) {
                if (view == getItemView()) {
                    if (BaseAdapter.this.mItemLongClickListener != null) {
                        return BaseAdapter.this.mItemLongClickListener.onItemLongClick(BaseAdapter.this.mRecyclerView, view, viewHolderPosition);
                    }
                    return false;
                } else if (!(BaseAdapter.this.mChildLongClickListeners == null || (onChildLongClickListener = (OnChildLongClickListener) BaseAdapter.this.mChildLongClickListeners.get(view.getId())) == null)) {
                    return onChildLongClickListener.onChildLongClick(BaseAdapter.this.mRecyclerView, view, viewHolderPosition);
                }
            }
            return false;
        }

        public final View getItemView() {
            return this.itemView;
        }

        public final <V extends View> V findViewById(int i) {
            return getItemView().findViewById(i);
        }
    }

    public void onAttachedToRecyclerView(RecyclerView recyclerView) {
        RecyclerView.LayoutManager generateDefaultLayoutManager;
        this.mRecyclerView = recyclerView;
        if (recyclerView.getLayoutManager() == null && (generateDefaultLayoutManager = generateDefaultLayoutManager(this.mContext)) != null) {
            this.mRecyclerView.setLayoutManager(generateDefaultLayoutManager);
        }
    }

    public void onDetachedFromRecyclerView(RecyclerView recyclerView) {
        this.mRecyclerView = null;
    }

    /* access modifiers changed from: protected */
    public RecyclerView.LayoutManager generateDefaultLayoutManager(Context context) {
        return new LinearLayoutManager(context);
    }

    public void setOnItemClickListener(OnItemClickListener onItemClickListener) {
        checkRecyclerViewState();
        this.mItemClickListener = onItemClickListener;
    }

    public void setOnChildClickListener(int i, OnChildClickListener onChildClickListener) {
        checkRecyclerViewState();
        if (this.mChildClickListeners == null) {
            this.mChildClickListeners = new SparseArray<>();
        }
        this.mChildClickListeners.put(i, onChildClickListener);
    }

    public void setOnItemLongClickListener(OnItemLongClickListener onItemLongClickListener) {
        checkRecyclerViewState();
        this.mItemLongClickListener = onItemLongClickListener;
    }

    public void setOnChildLongClickListener(int i, OnChildLongClickListener onChildLongClickListener) {
        checkRecyclerViewState();
        if (this.mChildLongClickListeners == null) {
            this.mChildLongClickListeners = new SparseArray<>();
        }
        this.mChildLongClickListeners.put(i, onChildLongClickListener);
    }

    private void checkRecyclerViewState() {
        if (this.mRecyclerView != null) {
            throw new IllegalStateException("are you ok?");
        }
    }
}

File: decompiled/sources/com/hjq/base/BaseAdapter.kt
package com.hjq.base

import android.content.Context
import android.util.SparseArray
import android.view.LayoutInflater
import android.view.View
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.hjq.base.action.ResourcesAction

abstract class BaseAdapter<VH : RecyclerView.ViewHolder>(context: Context?) :
    RecyclerView.Adapter<VH>(),
    ResourcesAction {
    private var mChildClickListeners: SparseArray<OnChildClickListener>? = null
    private var mChildLongClickListeners: SparseArray<OnChildLongClickListener>? = null
    private var mItemClickListener: OnItemClickListener? = null
    private var mItemLongClickListener: OnItemLongClickListener? = null
    private var mPositionOffset: Int = 0
    private var mRecyclerView: RecyclerView? = null
    private val mContext: Context = context ?: throw IllegalArgumentException("are you ok?")

    interface OnChildClickListener {
        fun onChildClick(recyclerView: RecyclerView?, view: View, position: Int)
    }

    interface OnChildLongClickListener {
        fun onChildLongClick(recyclerView: RecyclerView?, view: View, position: Int): Boolean
    }

    interface OnItemClickListener {
        fun onItemClick(recyclerView: RecyclerView?, view: View, position: Int)
    }

    interface OnItemLongClickListener {
        fun onItemLongClick(recyclerView: RecyclerView?, view: View, position: Int): Boolean
    }

    final override fun onBindViewHolder(holder: VH, position: Int) {
        mPositionOffset = position - holder.adapterPosition
        (holder as BaseAdapter<*>.ViewHolder).onBindView(position)
    }

    fun getRecyclerView(): RecyclerView? {
        return mRecyclerView
    }

    override fun getContext(): Context {
        return mContext
    }

    abstract inner class ViewHolder : RecyclerView.ViewHolder, View.OnClickListener, View.OnLongClickListener {
        constructor(layoutId: Int) : this(
            LayoutInflater.from(getContext()).inflate(layoutId, mRecyclerView, false)
        )

        constructor(view: View) : super(view) {
            if (mItemClickListener != null) {
                view.setOnClickListener(this)
            }
            if (mItemLongClickListener != null) {
                view.setOnLongClickListener(this)
            }

            val childClickListeners = mChildClickListeners
            if (childClickListeners != null) {
                for (i in 0 until childClickListeners.size()) {
                    val childView = findViewById<View>(childClickListeners.keyAt(i))
                    if (childView != null) {
                        childView.setOnClickListener(this)
                    }
                }
            }

            val childLongClickListeners = mChildLongClickListeners
            if (childLongClickListeners != null) {
                for (i in 0 until childLongClickListeners.size()) {
                    val childView = findViewById<View>(childLongClickListeners.keyAt(i))
                    if (childView != null) {
                        childView.setOnLongClickListener(this)
                    }
                }
            }
        }

        protected fun getViewHolderPosition(): Int {
            return layoutPosition + mPositionOffset
        }

        override fun onClick(view: View) {
            val viewHolderPosition = getViewHolderPosition()
            if (viewHolderPosition >= 0 && viewHolderPosition < itemCount) {
                if (view === getItemView()) {
                    mItemClickListener?.onItemClick(mRecyclerView, view, viewHolderPosition)
                } else {
                    val listener = mChildClickListeners?.get(view.id)
                    if (listener != null) {
                        listener.onChildClick(mRecyclerView, view, viewHolderPosition)
                    }
                }
            }
        }

        override fun onLongClick(view: View): Boolean {
            val viewHolderPosition = getViewHolderPosition()
            if (viewHolderPosition >= 0 && viewHolderPosition < itemCount) {
                if (view === getItemView()) {
                    val listener = mItemLongClickListener
                    return listener?.onItemLongClick(mRecyclerView, view, viewHolderPosition) ?: false
                }

                val childListener = mChildLongClickListeners?.get(view.id)
                if (childListener != null) {
                    return childListener.onChildLongClick(mRecyclerView, view, viewHolderPosition)
                }
            }
            return false
        }

        abstract fun onBindView(position: Int)

        fun getItemView(): View {
            return itemView
        }

        fun <V : View> findViewById(id: Int): V? {
            return getItemView().findViewById(id)
        }
    }

    override fun onAttachedToRecyclerView(recyclerView: RecyclerView) {
        mRecyclerView = recyclerView
        if (recyclerView.layoutManager == null) {
            val layoutManager = generateDefaultLayoutManager(mContext)
            if (layoutManager != null) {
                mRecyclerView!!.layoutManager = layoutManager
            }
        }
    }

    override fun onDetachedFromRecyclerView(recyclerView: RecyclerView) {
        mRecyclerView = null
    }

    protected open fun generateDefaultLayoutManager(context: Context): RecyclerView.LayoutManager? {
        return LinearLayoutManager(context)
    }

    fun setOnItemClickListener(listener: OnItemClickListener?) {
        checkRecyclerViewState()
        mItemClickListener = listener
    }

    fun setOnChildClickListener(id: Int, listener: OnChildClickListener?) {
        checkRecyclerViewState()
        if (mChildClickListeners == null) {
            mChildClickListeners = SparseArray()
        }
        mChildClickListeners!!.put(id, listener)
    }

    fun setOnItemLongClickListener(listener: OnItemLongClickListener?) {
        checkRecyclerViewState()
        mItemLongClickListener = listener
    }

    fun setOnChildLongClickListener(id: Int, listener: OnChildLongClickListener?) {
        checkRecyclerViewState()
        if (mChildLongClickListeners == null) {
            mChildLongClickListeners = SparseArray()
        }
        mChildLongClickListeners!!.put(id, listener)
    }

    private fun checkRecyclerViewState() {
        if (mRecyclerView != null) {
            throw IllegalStateException("are you ok?")
        }
    }
}
